### PR TITLE
Added url as a parameter for FTP objects

### PR DESCRIPTION
**Pull request by samuel-massinon**
Ready to review

### CODE REVIEW

**Comment by omus** [comment]
This kind of change makes me think we should move away from just using keywords. `url` can contain all the details provided from `host`, `ssl`, `user`, and `pswd`. The methods could look like:

``` julia
FTP(url::AbstractString) = ...
FTP(host::AbstractString, username, password, ssl=false) = ...
```

**Comment by samuel-massinon** [comment]
Hmm, I'm trying to figure out how this would work. There's still 

``` julia
    implicit::Bool
    verify_peer::Bool
    active_mode::Bool
```

So I would leave those as keywords.
A user that passes a `url` might still want to pass a username and password.
A user that passes a `url` and wants ssl _must_ pass `ssl=true`, unless we determine ssl from the `url`.
The only truly useless variable when a `url` is given is `hostname`.
I think it's easier and gives the user more options to leave as is. If they pass a `url` with a username and also pass a `username` in the parameter, it will still work but the parameter gets ignored. If we leave it as is I'll make note of it in the README/mkdocs.

**Comment by omus** [comment]
The URLs take the format of `protocol://username:password@hostname`. I think if we have a URL then the username and password should be retrieved from their. Otherwise we could get a username in both the URL or the username parameter. I think we can drop the `ssl` keyword when using a URL as `ftps` indicates FTP over SSL.

**Comment by omus** [comment]
You can definitely use keywords and I think you should use them where it makes sense. I just think there is some room to use multiple dispatch to make the interface better and using exclusively keywords means we're kinda stuck.

File: test/ftp_object.jl [hunk 1]
```diff
@@ -66,6 +66,13 @@ end
     close(ftp)
 end
 
+# @testset "connection with url" begin
+#     host = hostname(server)
+#     ftp = FTP(; url="ftp://$host/", opts...)
+#     @test ftp.ctxt.url == "ftp://$host/"
+#     close(ftp)
+# end
+
```
Comment (omus) [inline test/ftp_object.jl]: Is the test supposed to be commented out?

Comment (kano31) [reply test/ftp_object.jl]: Oops!

**Review by omus — APPROVED** [review]
Looks good. Let's see what the CI has to say

**Pull request merged**